Blish HUD/Controls/ColorBox.cs

```diff
@@ -1,114 +1,111 @@
-        //    };
-        //    spriteHighlight = Resources.Control.TextureAtlasControl.GetRegion("colorpicker/cp-clr-active");
-        //}
-
-        //#endregion
-
-        //private readonly int _drawVariation;
-
-        //public ColorBox() : base() {
-        //    LoadStatics();
-
-        //    this.Size = new Point(COLOR_SIZE);
-
-        //    _drawVariation = RandomUtil.GetRandom(0, 3);
-        //}
-
-        //protected override void OnMouseMoved(MouseEventArgs e) {
-        //    base.OnMouseMoved(e);
-
-        //    this.BasicTooltipText = this.Color?.Id.ToString() ?? "None";
-        //}
-
-        //protected override CaptureType CapturesInput() {
-        //    return CaptureType.Mouse;
-        //}
-
-        //protected override void Paint(SpriteBatch spriteBatch, Rectangle bounds) {
-        //    spriteBatch.Draw(spriteBoxes[_drawVariation], bounds, this.Color?.Fur?.Rgb.ToXnaColor() ?? Microsoft.Xna.Framework.Color.White);
-
-        //    if (this.MouseOver || this.Selected)
-        //        spriteBatch.Draw(spriteHighlight, bounds, Microsoft.Xna.Framework.Color.White * 0.7f);
-        //}
-
-    //}
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class ColorBox : Control {
+
+        public event EventHandler<EventArgs> OnColorChanged;
+        public event EventHandler<EventArgs> OnSelected;
+
+        private const int DEFAULT_COLOR_SIZE = 32;
+        private const string COLOR_CHANGE_SOUND_NAME = "audio/color-change";
+        private const string DRAW_VARIATION_VERSION_ONE_NAME = "colorpicker/cp-clr-v1";
+        private const string DRAW_VARIATION_VERSION_TWO_NAME = "colorpicker/cp-clr-v2";
+        private const string DRAW_VARIATION_VERSION_THREE_NAME = "colorpicker/cp-clr-v3";
+        private const string DRAW_VARIATION_VERSION_FOUR_NAME = "colorpicker/cp-clr-v4";
+        private const string HIGHLIGHT_NAME = "colorpicker/cp-clr-active";
+
+        private readonly int drawVariation;
+
+        private bool selected = false;
+
+        public bool Selected {
+            get => selected;
+            set {
+                if (SetProperty(ref selected, value)) {
+                    this.OnSelected?.Invoke(this, EventArgs.Empty);
+
+                    if (this.Visible)
+                        Content.PlaySoundEffectByName(COLOR_CHANGE_SOUND_NAME);
+                }
+            }
+        }
+
+        private Gw2Sharp.WebApi.V2.Models.Color color;
+
+        public Gw2Sharp.WebApi.V2.Models.Color Color {
+            get => color;
+            set {
+                if (color == value)
+                    return;
+
+                color = value;
+
+                OnPropertyChanged(nameof(this.ColorId));
+                OnPropertyChanged();
```

Comment:
I think this duplicate is a relic of old code that was commented out at different times.  This can all be replaced with:
```
set {
	if (SetProperty(ref color, value)) {
		this.ColorChanged?.Invoke(this, EventArgs.Empty);
	}
}
```

Reply:
But `OnPropertyChanged(nameof(this.ColorId));` still needs to be called or not?

Reply:
Oh, `SetProperty` actually builds that in.

https://github.com/blish-hud/Blish-HUD/blob/e8d9df02f71bfdcd2d8fa118b353138ce1db9889/Blish%20HUD/Controls/Control.cs#L802-L810

Reply:
As discussed in Discord, this was for the `ColorId` property if I had been paying attention.  We should be good to actually fully remove that property as it's just redundant.